cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -273,6 +322,15 @@ private void recordProgramSuspendResume(ProgramId programId, String pid, String
       .add(pid)
       .build();
     RunRecordMeta record = get(key, RunRecordMeta.class);
+
+    // Check without the version string only for default version
+    if (record == null && (programId.getVersion().equals(ApplicationId.DEFAULT_VERSION))) {
+      key = getVersionLessProgramKeyBuilder(fromType, programId)
```

Comment:
The `deleteAll(key)` below will delete `RunMeta` for all the versions with this key, right?